snap7/client_async.py

```diff
@@ -77,29 +81,17 @@ async def as_db_write(self, db_number, start, data, timeout=1):
         """
         This is the asynchronous counterpart of Cli_DBWrite with asyncio features.
         """
-        wordlen = snap7.types.S7WLByte
-        type_ = snap7.types.wordlen_to_ctypes[wordlen]
-        size = len(data)
-        cdata = (type_ * size).from_buffer_copy(data)
-        logger.debug(f"db_write db_number:{db_number} start:{start} size:{size} data:{data}")
-        check = self._library.Cli_AsDBWrite(self._pointer, db_number, start, size, byref(cdata))
-        request_in_time = await self.as_check_and_wait(timeout)
-        if request_in_time is False:
+        check = Client.as_db_write(self, db_number, start, data)
```

Comment:
see this already starts to look much cleaner, and if we want to switch to cffi at some point we only need to modify the client class.